### PR TITLE
Launch S3 on all text datasets.

### PR DESCRIPTION
**Pull request by us**
Except wikipedia because it uses beam.
Related issue: #737

### CODE REVIEW

File: tensorflow_datasets/text/definite_pronoun_resolution.py [hunk 1]
```diff
@@ -93,19 +100,23 @@ def _split_generators(self, dl_manager):
             gen_kwargs={'filepath': files['train']}),
     ]
 
+  @tfds.core.drop_key_if_not_s3
```
Comment (pierrot0) [inline tensorflow_datasets/text/definite_pronoun_resolution.py]: Can be removed now.

File: tensorflow_datasets/text/cnn_dailymail.py [hunk 2]
```diff
@@ -219,14 +225,14 @@ class CnnDailymail(tfds.core.GeneratorBasedBuilder):
           name='bytes',
           description=('Uses byte-level text encoding with '
                        '`tfds.features.text.ByteTextEncoder`'),
-          text_encoder_config=tfds.features.text.TextEncoderConfig(
+        text_encoder_config=tfds.features.text.TextEncoderConfig(
```
Comment (Conchylicultor) [inline tensorflow_datasets/text/cnn_dailymail.py]: Could you run `pylint` to format your workspace ? For instance why removing spaces here ?

File: tensorflow_datasets/text/cnn_dailymail.py [hunk 3]
```diff
@@ -249,7 +255,7 @@ def _info(self):
     )
 
   def _vocab_text_gen(self, paths):
-    for ex in self._generate_examples(paths):
+    for fname, ex in self._generate_examples(paths):
```
Comment (Conchylicultor) [inline tensorflow_datasets/text/cnn_dailymail.py]: If the variable is unused, then you can use `_`:

```
for _, ex in self._generate_examples(paths):
```

File: tensorflow_datasets/text/definite_pronoun_resolution.py [hunk 4]
```diff
@@ -49,11 +49,18 @@
...
+  # 2.0.0: S3 with new hashing function (different shuffle).
+  # 1.0.0: S3 (new shuffling, sharding and slicing mechanism).
   BUILDER_CONFIGS = [
       tfds.core.BuilderConfig(
           name='plain_text',
           version=tfds.core.Version(
               '0.0.1', experiments={tfds.core.Experiment.S3: False}),
+          supported_versions=[
+            tfds.core.Version("2.0.0"),
+            tfds.core.Version("1.0.0"),
```
Comment (Conchylicultor) [inline tensorflow_datasets/text/definite_pronoun_resolution.py]: You should only add a single new version, as only the new hash function is supported now.

File: tensorflow_datasets/text/cnn_dailymail.py (hunk 2 above)
Comment (us) [reply tensorflow_datasets/text/cnn_dailymail.py]: I run and fixed

**Pull request merged**